### ISSUE TITLE
Keras Imported CNN - dimensionality problem between `embedding` and `conv1d`

### ISSUE THREAD

**Issue by winobes** [deeplearning4j/deeplearning4j#4466]
#### Issue Description

[gist](https://gist.github.com/winobes/d69e5ba1b0571290a82193e050c3e9c6)

In `test` I get an error from `RnnToFeedForwardPreProcessor` saying the input is expected to be rank 3. I pulled apart the layers in `testLayers` and it looks like the problem is happening between the embedding and the convolution layers.
 
@maxpumperla Adam and Raver were speculating that the Keras conv1d layer might do some automatic reshaping of the output from the embedding layer. But they said you'd know.

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: 0.9.2
* platform information (OS, etc): Ubuntu 16.04
* CUDA version: No CUDA

**Comment by maxpumperla** [comment]
hi @winobes, yeah, I remember you from gitter. thanks for getting around to reporting this.

here's the issue: keras embedding layers always output 3d shape, dl4j embedding layers are feed-forward layers, so you need input preprocessors to connect them to RNN layers. Here's an example, if you're interested:

https://github.com/deeplearning4j/deeplearning4j/blob/25394159a0bbe3370d4928506cac3876a44b8704/deeplearning4j-core/src/test/java/org/deeplearning4j/nn/layers/feedforward/embedding/EmbeddingLayerTest.java#L171-L172

In your particular example, the time-series length is specified by `input_length` in the keras model, i.e. `136`. Upon import, we need to pick up this information and build an according `FeedForwardToRnnPreProcessor` with it. I'm on it!

**Comment by EricSteinberger** [comment]
Has the support for that been added? If not, when will it be? We need that for project in our firm. Would reeeeally appreciate if that could be done!

**Comment by maxpumperla** [comment]
@EricSteinberger is your use case the same, i.e. Embedding - CNN1D or do you want to do e.g. Embedding into LSTM?

**Comment by comoq3c** [comment]
right now, I am feeding the output from embedding layer to conv1d, and run into the same issue. Really appreciate your consideration working on this

**Comment by comoq3c** [comment]
but I do not use BatchNormalization in my model

**Comment by agibsonccc** [comment]
Hey folks - let @maxpumperla comment on this. Thanks!

**Comment by comoq3c** [comment]
is there any update on this?

**Comment by maxpumperla** [comment]
@comoq3c @winobes @EricSteinberger we have a unit test for this on master now (after merging #4686):

https://github.com/deeplearning4j/deeplearning4j/blob/8a3311c1fb00ed0d90328b35cdd5548585d203c2/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/weights/KerasWeightSettingTests.java#L109-L128

I'm closing this for now. Let me know if you encounter problems. thanks!
